NDTensors/src/lib/UnallocatedArrays/src/unallocatedfill.jl

```diff
@@ -0,0 +1,73 @@
+using FillArrays: FillArrays, AbstractFill, Fill, broadcasted_fill, kron_fill, mult_fill
+using NDTensors.SetParameters: Position, set_parameters
+
+struct UnallocatedFill{ElT,N,Axes,Alloc} <: AbstractFill{ElT,N,Axes}
+  f::Fill{ElT,N,Axes}
+  alloc::Alloc
+
+  function UnallocatedFill{ElT,N,Axes}(
+    f::Fill, alloc::Type{<:AbstractArray{ElT,N}}
+  ) where {ElT,N,Axes}
+    return new{ElT,N,Axes,Type{alloc}}(f, alloc)
+  end
+end
+
+function UnallocatedFill{ElT,N}(f::Fill, alloc) where {ElT,N}
+  return UnallocatedFill{ElT,N,typeof(axes(f))}(f, alloc)
+end
+
+function UnallocatedFill{ElT}(f::Fill, alloc) where {ElT}
+  return UnallocatedFill{ElT,ndims(f)}(f, alloc)
+end
+
+function UnallocatedFill(f::Fill, alloc)
+  return UnallocatedFill{eltype(f)}(f, alloc)
+end
+
+set_alloctype(f::Fill, alloc::Type{<:AbstractArray}) = UnallocatedFill(f, alloc)
+
+Base.parent(F::UnallocatedFill) = F.f
+
+Base.convert(::Type{<:UnallocatedFill}, A::UnallocatedFill) = A
+
+#############################################
+# Arithmatic
+
+# mult_fill(a, b, val, ax) = Fill(val, ax)
+function FillArrays.mult_fill(a::UnallocatedFill, b, val, ax)
+  return UnallocatedFill(Fill(val, ax), alloctype(a))
+end
+FillArrays.mult_fill(a, b::UnallocatedFill, val, ax) = mult_fill(b, a, val, ax)
+function FillArrays.mult_fill(a::UnallocatedFill, b::UnallocatedFill, val, ax)
+  @assert(alloctype(a) == alloctype(b))
```

Comment:
```suggestion
  @assert alloctype(a) == alloctype(b)
```
Same thing elsewhere.

Reply:
Fixed across the board